llama-index-core/llama_index/core/readers/file/base.py

```diff
@@ -14,12 +15,38 @@
 from fsspec.implementations.local import LocalFileSystem
 from typing import Any, Callable, Dict, Generator, List, Optional, Type
 
-from llama_index.core.readers.base import BaseReader
+from llama_index.core.readers.base import BaseReader, BaseResourcesReader
 from llama_index.core.async_utils import run_jobs, get_asyncio_module
 from llama_index.core.schema import Document
 from tqdm import tqdm
 
 
+class BaseFilesystemReader(BaseResourcesReader, ABC):
+    @abstractmethod
+    def read_file_content(self, input_file: Path, **kwargs) -> bytes:
```

Comment:
Kind of curious how this differs from `load_resource()` -- feels like it would be the same method?

Reply:
I guess this returns bytes